File: examples/quickstart/python/pulserpc/rpc.py
"""RPC error handling for JSON-RPC 2.0"""

from typing import Any


class RPCError(Exception):
    """Exception class for JSON-RPC errors"""
    
    def __init__(self, code: int, message: str, data: Any = None):
        self.code = code
        self.message = message
        self.data = data
        super().__init__(f"RPCError {code}: {message}")

File: pkg/runtime/runtimes/ts-node/pulserpc/rpc.ts
/**
 * RPC error handling for JSON-RPC 2.0
 */

export class RPCError extends Error {
  public code: number;
  public data?: any;

  constructor(code: number, message: string, data?: any) {
    super(message);
    this.name = `RPCError ${code}`;
    this.code = code;
    this.data = data;
    if (Error.captureStackTrace) {
      Error.captureStackTrace(this, RPCError);
    }
  }
}
